cbsurge/az/blobstorage.py

```diff
@@ -0,0 +1,250 @@
+
+    util.validate_azure_storage_path(a_path=src_path)
+
+    proto, account_name, src_blob_path = src_path.split(':')
+    container_name, *src_path_parts = src_blob_path.split(os.path.sep)
+    rel_src_blob_path = os.path.sep.join(src_path_parts)
+    async with Session() as session:
+        async with session.get_blob_container_client(account_name=account_name,container_name=container_name) as cc:
+            blob_client = cc.get_blob_client(blob=rel_src_blob_path)
+            return await download(blob_client=blob_client, dst_path=dst_path )
+
+
+async def upload(src_path, dst_path, blob_client:BlobClient = None, blob_type:BlobType =  BlobType.BLOCKBLOB, overwrite=True):
+    """
+
+    :param src_path: str, path to the local file to be uploaded
+    :param dst_path: str, the fully qualified path to a blob in az in format az:{account}:{container}/path.ext
+    :param blob_client: instance of Azure Storage  BlobClient
+    :param blob_type: one of BlobType.APPENDBLOB, BlobType.BLOCKBLOB, BlobType.PAGEBLOB
+    :param overwrite:bool, if the blob will be overwritten
+    :return: str, the fully qualified path to a blob in az in format az:{account}:{container}/path.ext
+    """
+    if not os.path.exists(src_path):
+        raise ValueError(f"File {src_path} does not exist")
+    try:
+        logger.debug(f"Uploading {src_path} to {dst_path}")
+        with open(src_path, mode="rb") as data:
+            await blob_client.upload_blob(
+                data=data,
+                overwrite=overwrite,
+                blob_type = blob_type,
+                #max_concurrency=8
+            )
+    except Exception as e:
+        logger.error(f"Failed to upload {dst_path} from {src_path}: {e}")
+    else:
+        logger.debug(f"Blob {src_path} was successfully uploaded to {dst_path}")
+        return dst_path
+
+
+
+
+
+
+async def upload_blob(src_path: str = None, dst_path: str = None):
+    """
+    Upload a blob from a local file
+    :param src_path: str, path to the local file to be uploaded
+    :param dst_path: str, the fully qualified path to a blob in az in format az:{account}:{container}/path.ext
+    :return: str, the fully qualified path to a blob in az in format az:{account}:{container}/path.ext
+
+    """
+    assert isinstance(src_path, str), f'src_path must be a valid string, not {type(src_path)}'
+    util.validate_azure_storage_path(a_path=dst_path)
+    proto, account_name, dst_blob_path = dst_path
+    container_name, *dst_path_parts = dst_blob_path.split(os.path.sep)
+    rel_dst_blob_path = os.path.sep.join(dst_path_parts)
+    async with Session() as session:
+        with session.get_blob_container_client(account_name=account_name,container_name=container_name) as cc:
+            blob_client = cc.get_blob_client(blob=rel_dst_blob_path)
+            return upload(src_path, dst_path, blob_client)
+
+async def upload_blob_with_session(session:Session=None, src_path:str=None, dst_path: str = None):
```

Comment:
Can this be removed? I think `upload_blob` is enough if new Session class always return same instance now.